multistateview/src/main/java/me/xdj/view/MultiStateView.java

```diff
@@ -0,0 +1,208 @@
+        this(context, attrs, "");
+        mHandler = new EventHandler(EventRunner.getMainEventRunner());
+    }
+
+    /**
+     * MultiStateView.
+     *
+     * @param context context.
+     * @param attrs attrset.
+     * @param defStyleAttr defattr.
+     */
+    public MultiStateView(Context context, AttrSet attrs, String defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        mHandler = new EventHandler(EventRunner.getMainEventRunner());
+    }
+
+    @Override
+    public void addComponent(Component child) {
+        validContentView(child);
+        super.addComponent(child);
+    }
+
+    @Override
+    public void addComponent(Component child, int index) {
+        validContentView(child);
+        super.addComponent(child, index);
+    }
+
+    @Override
+    public void addComponent(Component child, int width, int height) {
+        validContentView(child);
+        super.addComponent(child, width, height);
+    }
+
+    @Override
+    public void addComponent(Component child, ComponentContainer.LayoutConfig params) {
+        validContentView(child);
+        super.addComponent(child, params);
+    }
+
+    @Override
+    public void addComponent(Component child, int index, ComponentContainer.LayoutConfig params) {
+        validContentView(child);
+        super.addComponent(child, index, params);
+    }
+
+    /**
+     * Change view state.
+     *
+     * @param state State type.
+     */
+    public void setViewState(int state) {
+        if (state != mCurrentState) {
+            Component view = getView(state);
+            getCurrentView().setVisibility(HIDE);
+            mCurrentState = state;
+            if (view != null) {
+                view.setVisibility(VISIBLE);
+            } else {
+                int resLayoutId = mLayoutIdArray.get(state).get();
+                view = LayoutScatter.getInstance(getContext()).parse(resLayoutId, this, false);
+                mStateViewArray.put(state, view);
+                addComponent(view);
+                view.setVisibility(VISIBLE);
+
+                if (mOnInflateListener != null) {
+                    mOnInflateListener.onInflate(state, view);
+                }
+            }
+        }
+    }
+
+    /**
+     * get current status view.
+     *
+     * @return currentState.
+     */
+    public int getViewState() {
+        return mCurrentState;
+    }
+
+    /**
+     * Get the specified status View.
+     *
+     * @param state state.
+     * @return Specified state View.
+     */
+    public Component getView(int state) {
+        if (mStateViewArray.get(state).isPresent()) {
+            return mStateViewArray.get(state).get();
+        }
+        return null;
+    }
+
+    /**
+     * getCurrentView.
+     *
+     * @return currentStateView.
+     */
+    public Component getCurrentView() {
+        Component view = getView(mCurrentState);
+        if (view == null && mCurrentState != STATE_CONTENT) {
```

Comment:
All scenarios are not handled here. 
Check this scenario : When "view" is null and also mCurrentState is equal to STATE_CONTENT - this scenario is not handled in this code